satellite/satellitedb/satellitedbtest/cockroachnamespace_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedbtest_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"storj.io/storj/private/dbutil"
+	"storj.io/storj/private/dbutil/pgutil/pgtest"
+	"storj.io/storj/private/testcontext"
+	dbx "storj.io/storj/satellite/satellitedb/dbx"
+	"storj.io/storj/satellite/satellitedb/satellitedbtest"
+)
+
+func TestNewCockroach(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	if *pgtest.CrdbConnStr == "" {
+		t.Skip("Cockroachdb flag missing")
+	}
+	namespacedDBName := "namespacedTestDB"
+	testdb, err := satellitedbtest.NewCockroach(zap.L(), namespacedDBName)
+	require.NoError(t, err)
+
+	// assert new test db exists
+	driver, source, err := dbutil.SplitConnstr(*pgtest.CrdbConnStr)
+	if err != nil {
+		t.Fatal(err)
+	}
+	db, err := dbx.Open(driver, source)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var exists *bool
+	row := db.QueryRow(fmt.Sprintf(`SELECT EXISTS (
```

Comment:
Why not use parameter binding here, and on line 56?

```go
    row := db.QueryRow(`SELECT EXISTS (
            SELECT datname FROM pg_catalog.pg_database WHERE lower(datname) = lower($1)
        );`, namespacedDBName)
```

Safer _and_ simpler, in this case at least, unless it breaks for some reason.

Reply:
How come when I do this, https://github.com/storj/storj/pull/3646/files#diff-cafd8c613e35419698a24aaa6a1d3929R37:
```
_, err = db.Exec("CREATE DATABASE $1;", pq.QuoteIdentifier(namespacedTestDB))
```
I get this error:
```
pq: at or near "1": syntax error
```

That is why I started using the `fmt.Sprintf`, but I'm realizing now that is now necessary with  `db.QueryRow()` so I will remove it.

Reply:
Ah, that's because CREATE DATABASE takes an identifier instead of a string literal. PG/cockroach doesn't see that as a valid place to put a parameter.

Schema names, table names, database names, and column names are all "identifiers". They have to be quoted with double-quotes when you want to escape them, as opposed to single-quoted string literals.

Reply:
ah interesting. thanks for the explanation.